pipdeptree.py

```diff
@@ -599,14 +605,21 @@ def main():
         if args.warn == 'fail' and (conflicting or cyclic):
             return_code = 1
 
-    show_only = set(args.packages.split(',')) if args.packages else None
+    show_only = set(args.packages.split(',')) if args.packages else set()
```

Comment:
Since the variable name is `show_only`, I think it would be better to have its default value to be `None` and not empty set. On the other hand, it's ok for `exclude` to have `set()` as the default value because absence of `--exclude`, translates to "no packages should be excluded"

The condition to show error and exit can be modified as follows,

```
show_only and (show_only & exclude)
```

Reply:
Thanks, didn't realize python stops evaluating once a condition is not met 👍